### PR TITLE
Update Mac download to point to new codesigned Sage app

### PR DESCRIPTION
**Pull request by NathanDunfield**
To go with https://trac.sagemath.org/ticket/31638

### CODE REVIEW

**Comment by NathanDunfield** [comment]
Paging @mkoeppe @culler @slel @unhyperbolic.

File: src/download-mac.html [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
 {% block content %}
 
- <h1 class="narrow">Download Binaries for Apple Mac OSX</h1>
+ <h1 class="narrow">Download Binaries for Apple's macOS</h1>
 
  <div class="txt narrow">
-  These binaries are for OS X version 10.4 and higher.
-  Usually the binary for a version or two before yours should work, e.g. 10.7 should work on 10.8.
-  For more information <a href="http://files.sagemath.org/osx/README.txt">read the installation instructions</a>.
+   These binaries are all recent versions of macOS.  Usually the
```
Comment (mkoeppe) [inline src/download-mac.html:10]: typo: are FOR all recent versions

**Comment by mkoeppe** [comment]
LGTM except for the typo noted above.

File: src/download-mac.html (hunk 1 above)
Comment (NathanDunfield) [reply src/download-mac.html]: Thanks, fixed.

**Review by mkoeppe — APPROVED** [review]

**Pull request merged**